components/camel-rest-swagger/src/main/java/org/apache/camel/component/rest/swagger/RestSwaggerComponent.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.rest.swagger;
+
+import java.util.Map;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.impl.DefaultComponent;
+import org.apache.camel.spi.RestProducerFactory;
+
+/**
+ * An awesome REST component backed by Swagger specifications. Creates endpoints that connect to REST APIs defined by
+ * Swagger specification. This component delegates to other {@link RestProducerFactory} components to act as REST
+ * clients, but it configures them from Swagger specification. Client needs to point to operation that it wants to
+ * invoke via REST, provide any additional HTTP headers as headers , and payload as the body of the incoming message.
+ * <p>
+ * Example usage using Java DSL:
+ * <p>
+ *
+ * <pre>
+ * from("rest-swagger:getPetById?endpoint=https://api.petstore.com").to(...)
+ * </pre>
+ *
+ * This relies on only one {@link RestProducerFactory} component being available to Camel, you can use specific, for
+ * instance preconfigured component by using the {@code componentName} endpoint property. For example using Undertow
+ * component in Java DSL:
+ * <p>
+ *
+ * <pre>
+ * Component petstore = new UndertowComponent();
+ * petstore.setSslContextParameters(...);
+ * //...
+ * camelContext.addComponent("petstore", petstore);
+ *
+ * from("rest-swagger:getPetById?endpoint=https://api.petstore.com&componentName=petstore").to(...)
+ * </pre>
+ */
+public class RestSwaggerComponent extends DefaultComponent {
```

Comment:
It would maybe be nice to be able to configure the endpoint on the component level so you dont have to specify that in the uri, eg that api.petstore.com thingy.